lib/components/chain_avatar.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+/// Chain avatar
+///
+/// This class renders http and asset, image and svg, or just returns a short name
+/// with a colored background
+class ChainAvatar extends StatelessWidget {
+  final String name;
+  final String? image;
+  final double size;
+
+  const ChainAvatar({
+    super.key,
+    required this.name,
+    this.image,
+    required this.size,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ClipRRect(
+      borderRadius: BorderRadius.circular(size / 2),
+      child: Container(
+        width: size,
+        height: size,
+        child: _buildContent(context),
+      ),
+    );
+  }
+
+  Widget _buildContent(BuildContext context) {
+    if (image != null) {
```

Comment:
all of this logic can be moved away from the UI into the bloc.

Reply:
But which block? This should be a reusable component that can show any sort of image, svg, asset, png, web... 

I don't want to have to care about what asset it is, and I don't want to put that in every bloc that uses it.

Reply:
I just wanna show an image, IMO that's what the normal image widget could do it it wanted to, dump any kind of resource on it, and it works.

Reply:
if (image!.startsWith("http")) {
        if (image!.endsWith(".svg")) {
          return SvgPicture.network(image!, fit: BoxFit.cover);
        } else {
          return CachedNetworkImage(imageUrl: image!, fit: BoxFit.cover);
        }
      } else if (image!.startsWith("asset")) {
        if (image!.endsWith(".svg")) {
          return SvgPicture.asset(image!, fit: BoxFit.scaleDown);
        } else {
          return Image.asset(image!);
        }
      }
      
      ALl these stuff can be  moved away from the UI. You are doing logic in the UI and this should be avoided. 
      
"But which block? This should be a reusable component that can show any sort of image, svg, asset, png, web..."
Then have the reusable component take a Widget instead of an Image, this way you can pass in there anything you want. 

It is not up to the UI component to determine if its HTTP or asset or w.e, the UI component should be dumb, you give me a widget i render a widget.

Reply:
its fine here, this is a small scale project and this wont change. But when we write UI components, there is always a way to move most if not all logic away from them.